base/shared/src/main/scala/scalaz/typeclass/Magma.scala

```diff
@@ -0,0 +1,12 @@
+package scalaz
+package typeclass
+
+/* A Magma is a Set with a binary operation that returns a member of the same Set
+ * */
+trait Magma[A] {
+  def append(a1: A, a2: => A): A
+}
```

Comment:
Surely this is lawless. What good does it do us to have a `Magma` that's not a `Semigroup`?